src/lib/mlang.js

```diff
@@ -73,72 +75,182 @@ foam.CLASS({
 });
 
 
+/**
+ * Base class for all mLang queries.
+ *
+ * Contains many default implementations of methods.
+ */
 foam.CLASS({
-  package: 'foam.mlang',
-  name: 'Expr'
+  package: 'foam.mlang.predicate',
+  name: 'Expr',
+  abstract: true,
+  // PORT: There should be several interfaces here eventually, eg.
+  // Functor/Projectable/whatever we call f()'s interface.
+  // implements: ['ExprProtocol'],
```

Comment:
Delete this for now.  ExprProtocol was a thing added for swift, we should be able to do this better.